UniversityApp.BussinessLogic/Services/StudentsService.cs

```diff
@@ -99,34 +99,35 @@ public async Task<List<GradeDomainModel>> GetStudentGrades(int id)
             return result;
         }
 
-        public async Task<int?> AddStudentGrade(int id, GradeDomainModel grade)
+        public async Task<int?> AddStudentGrade(GradeDomainModel grade)
         {
-            var student = await _studentsRepository.GetStudent(id);
-
-            if(student == null)
+            if (!(await DoesStudentExists(grade.StudentId)))
             {
                 return null;
             }
 
             var gradeToAdd = _mapper.Map<Grade>(grade);
 
-            gradeToAdd.StudentId = id;
-
             var result = await _gradesRepository.AddGrade(gradeToAdd);
 
             return result;
         }
 
-        public async Task<bool> UpdateStudentGarde(int id, int gradeId, GradeDomainModel grade)
+        public async Task<bool> UpdateStudentGarde(int gradeId, GradeDomainModel grade)
         {
             var gradeToUpdate = _mapper.Map<Grade>(grade);
 
             gradeToUpdate.Id = gradeId;
-            gradeToUpdate.StudentId = id;
 
             var result = await _gradesRepository.UpdateGrade(gradeToUpdate);
 
             return result;
         }
+
+        public async Task<bool> DoesStudentExists(int id)
+        {
+            var exists = await _studentsRepository.DoesStudentExist(id);
+            return exists;
```

Comment:
I don't like this name. If more you want to use naming that describes what is returned then better name would be `doesStudentExist` or just `result` would be enough or `return await _studentsRepository.DoesStudentExist(id);`

Reply:
sure, done :)